### PR TITLE
fix ggplot

### PR DESCRIPTION
**Pull request by Qi77Qi**
tested in https://github.com/DataBiosphere/leonardo/pull/1189

### CODE REVIEW

File: terra-jupyter-python/Dockerfile [hunk 1]
```diff
@@ -28,21 +28,25 @@ RUN pip3 -V \
...
+ && pip3 install google-cloud-storage==1.23.0 \
+ && pip3 install --ignore-installed firecloud==0.16.25 \
  && pip3 install scikit-learn==0.20.0 \
  && pip3 install statsmodels==0.9.0 \
  && pip3 install ggplot==0.11.5 \
  && sed -i 's/pandas.lib/pandas/g' /usr/local/lib/python3.7/dist-packages/ggplot/stats/smoothers.py \
+ # the next few `sed` lines are workaround for a ggplot bug. See https://github.com/yhat/ggpy/issues/662
+ && sed -i 's/pandas.tslib.Timestamp/pandas.Timestamp/g' /usr/local/lib/python3.7/dist-packages/ggplot/stats/smoothers.py \
+ && sed -i 's/pd.tslib.Timestamp/pd.Timestamp/g' /usr/local/lib/python3.7/dist-packages/ggplot/stats/smoothers.py \
+ && sed -i 's/pd.tslib.Timestamp/pd.Timestamp/g' /usr/local/lib/python3.7/dist-packages/ggplot/utils.py \
```
Comment (Qi77Qi) [inline terra-jupyter-python/Dockerfile]: these are the fixes for ggplot

File: terra-jupyter-python/Dockerfile [hunk 2]
```diff
@@ -28,21 +28,25 @@ RUN pip3 -V \
...
- && pip3 install google-cloud-bigquery-datatransfer==0.1.1 \
- && pip3 install google-cloud-datastore==1.7.0 \
- && pip3 install google-cloud-resource-manager==0.28.1 \
- && pip3 install google-cloud-storage==1.13.0 \
- && pip3 install --ignore-installed firecloud==0.16.18 \
+ && pip3 install google-cloud-bigquery-datatransfer==0.4.1 \
+ && pip3 install google-cloud-datastore==1.10.0 \
+ && pip3 install google-cloud-resource-manager==0.30.0 \
+ && pip3 install google-cloud-storage==1.23.0 \
+ && pip3 install --ignore-installed firecloud==0.16.25 \
```
Comment (Qi77Qi) [inline terra-jupyter-python/Dockerfile]: upgrade google-cloud-xxx to latest

File: terra-jupyter-python/Dockerfile [hunk 3]
```diff
@@ -28,21 +28,25 @@ RUN pip3 -V \
...
  && sed -i 's/pandas.lib/pandas/g' /usr/local/lib/python3.7/dist-packages/ggplot/stats/smoothers.py \
+ # the next few `sed` lines are workaround for a ggplot bug. See https://github.com/yhat/ggpy/issues/662
+ && sed -i 's/pandas.tslib.Timestamp/pandas.Timestamp/g' /usr/local/lib/python3.7/dist-packages/ggplot/stats/smoothers.py \
+ && sed -i 's/pd.tslib.Timestamp/pd.Timestamp/g' /usr/local/lib/python3.7/dist-packages/ggplot/stats/smoothers.py \
+ && sed -i 's/pd.tslib.Timestamp/pd.Timestamp/g' /usr/local/lib/python3.7/dist-packages/ggplot/utils.py \
  && pip3 install bokeh==1.0.0 \
  && pip3 install pyfasta==0.5.2 \
  && pip3 install markdown==2.4.1 \
- && pip3 install pdoc==0.3.2 \
+ && pip3 install pdoc3==0.7.2 \
```
Comment (Qi77Qi) [inline terra-jupyter-python/Dockerfile]: pdoc doesn't support python 3...pdoc 3 does

File: terra-jupyter-python/Dockerfile [hunk 4]
```diff
@@ -73,7 +77,7 @@ RUN pip3 -V \
  && pip3 install intel-openmp==2018.0.0 \
  && pip3 install mkl==2018.0.3 \
  && pip3 install readline==6.2 \
- && pip3 install setuptools==36.4.0 \
+ && pip3 install setuptools==42.0.2 \
```
Comment (Qi77Qi) [inline terra-jupyter-python/Dockerfile]: six and setuptools are upgraded due to some google-cloud-xxx lib requires higher version

File: terra-jupyter-python/Dockerfile (hunk 1 above)
Comment (kyuksel) [reply terra-jupyter-python/Dockerfile]: Are those the strictest patterns to match? Like would ```'s/^pandas.tslib.Timestamp$/^pandas.Timestamp$/g'``` or something stricter be better?

Comment (Qi77Qi) [reply terra-jupyter-python/Dockerfile]: I think anywhere pandas.tslib.Timestamp is called will need to be replaced

**Pull request merged**